swarm/storage/mru/resourceviewid.go

```diff
@@ -0,0 +1,101 @@
+package mru
+
+import (
+	"encoding/json"
+	"hash"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+)
+
+// ResourceViewID represents a particular user's view of a resource ID
+type ResourceViewID struct {
+	resourceID ResourceID     `json:"resourceId"`
+	ownerAddr  common.Address `json:"ownerAddr"`
+}
+
+type ViewIDAddr []byte
+
+// ResourceViewID layout:
+// ResourceIDLength bytes
+// ownerAddr common.AddressLength bytes
+const resourceViewIDLength = ResourceIDLength + common.AddressLength
+
+// ResourceAddr calculates the resource owner id
+func (u *ResourceViewID) ResourceViewIDAddr() ViewIDAddr {
```

Comment:
The `Addr` part of this name is ambiguous. Can we specify that it's a swarm address somehow?